### DISCUSSION TITLE
On the fly gaps/margins

### DISCUSSION THREAD

**Discussion by Barbaross93** [qtile/qtile#2375] [Q&A] [answered]
Hi everyone,
I was wondering if there's a way to adjust `bar.Gap` or layout margins on the fly? @m-col on IRC suggested writing a function to do this. Unfortunately my python skills aren't great, but this is what I have so far:
```python
def gaps(qtile, side, direction, value):
	if (direction == "+"):
		old_gap = #Gets current gap size
		new_gap = old_gap + value
		Screen(side = bar.Gap(new_gap))
		# Refresh layout
	elif (direction == "-"):
		old_gap = #retrieve current size
		new_gap = old_gap - value
		Screen(side = bar.Gap(new_gap))
		#Refresh layout
```
followed by some lazy functions for specific gap sizes. I know the `side` variable won't work the way it is, I just wasn't sure how to implement it correctly. As for adjustable margins, I have no idea where to begin with that. Any help with achieving this functionality would be greatly appreciated!

**Comment by m-col** [comment, marked as answer]
I think you can modify `gap.size` directly.

Let's say you want to bind a key to increase the gap sizes by 10 pixels, you can change `gap.size` values for the current screen, and then probably just reconfigure the group with `group.layout_all`:
```py
from libqtile import lazy

@lazy.function
def increase_gaps(qtile):
    for gap in qtile.current_screen.gaps:
        gap.size += 10
    qtile.current_group.layout_all()
```
Then bind that function you do other lazy things. 

NB untested! But maybe this works?

**Reply by Barbaross93** [reply]
Nice, that does work! Thanks @m-col! Is it possible to specify a specific side? I could see:
```
from libqtile import lazy

@lazy.function
def increase_gaps(qtile):
    for gap in qtile.current_screen.gaps:
        if (gap == "left"):
          gap.size += 10
    qtile.current_group.layout_all()
```
But that seems terribly inefficient

**Reply by m-col** [reply]
Awesome. The gaps can be accessed via the attributes `screen.left` etc, i.e. `screen.left.size += 10`

**Reply by Barbaross93** [reply]
So like this?
```
@lazy.function
def increase_gaps(qtile):
    screen.left.size += 10
    qtile.current_group.layout_all()
```
Unfortunately that doesn't seem to work :(

**Reply by m-col** [reply]
Replace `screen` with `qtile.current_screen` -- that will be giving an error saying that `screen is not defined`

**Reply by Barbaross93** [reply]
Perfect! Thank you! This works great within `config.py`, but I was also hoping to get this in a separate python script like this:
```
#!/usr/bin/python

from libqtile import layout, bar, widget, hook
from libqtile.lazy import lazy


@lazy.function
def open_gap(qtile):
    qtile.current_screen.bottom.size += 400
    qtile.current_group.layout_all()

lazy.open_gap()
```
Unfortunately this doesn't work. Any ideas?

**Comment by Barbaross93** [comment]
It looks like I can affect the current layout's margins with `qtile.current_layout.margin += 18` followed by the refresh. However, I'd like to effect all layout's margins. Is this possible?

**Reply by m-col** [reply]
I think you'd have to iterate through `qtile.config.layouts` (Or I guess, `layouts`, as it's probably directly accessible from the config) and change them all manually.
